src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -455,6 +430,44 @@ private Collection<FunctionDescriptor> ReadFunctions(ScriptHostConfiguration con
             return ReadFunctions(metadatas, descriptorProviders);
         }
 
+        /// <summary>
+        /// Determines which script should be considered the "primary" entry point script.
+        /// </summary>
+        internal static string DeterminePrimaryScriptFile(JObject functionConfig, string[] functionFiles)
+        {
+            if (functionFiles.Length == 1)
+            {
+                // if there is only a single file, that file is primary
+                return functionFiles[0];
```

Comment:
I mean we'd be better off if we had the same requirements in the 1-source-file case as in the n-source-files case. Of course, changing it now is a breaking chance, but we should still consider it. Simpler rules, less magical, less cliff.